src/context/Queue.ts

```diff
@@ -33,6 +33,7 @@ interface Operation {
     failureMessage: string
     status?: Status
     type?: ActionType
+    data?: string
```

Comment:
I think this should be named differently than the `data` on line 133 (output of the operation when it succeeds). Maybe this could be like `metadata`?

I also think it would be better as an object so we can extend it in the future if we need to.